astropy/utils/shapes.py

```diff
@@ -74,35 +89,35 @@ def reshape(self, *args, **kwargs):
         the shape attribute (note: this may not be implemented for all classes
         using ``NDArrayShapeMethods``).
         """
-        return self._apply("reshape", *args, **kwargs)
+        return self._apply("reshape", *newshape, order=order)
 
-    def ravel(self, *args, **kwargs):
+    def ravel(self, order: Literal["C", "F", "A", "K"] = "C") -> Self:
         """Return an instance with the array collapsed into one dimension.
 
         Parameters are as for :meth:`~numpy.ndarray.ravel`. Note that it is
         not always possible to unravel an array without copying the data.
         If you want an error to be raise if the data is copied, you should
         should assign shape ``(-1,)`` to the shape attribute.
         """
-        return self._apply("ravel", *args, **kwargs)
+        return self._apply("ravel", order=order)
 
-    def flatten(self, *args, **kwargs):
+    def flatten(self, order: Literal["C", "F", "A", "K"] = "C") -> Self:
         """Return a copy with the array collapsed into one dimension.
 
         Parameters are as for :meth:`~numpy.ndarray.flatten`.
         """
-        return self._apply("flatten", *args, **kwargs)
+        return self._apply("flatten", order=order)
 
-    def transpose(self, *args, **kwargs):
+    def transpose(self, *axes: int | Sequence[int] | None) -> Self:
```

Comment:
Updating `transpose()` like you suggest would break at least https://github.com/astropy/astropy/blob/16786add79acbd89743b006d5eccf7d270089047/astropy/coordinates/tests/test_representation.py#L1606

Reply:
Admittedly the annotation I wrote is a bit sloppy, a more correct one would be
```python
@overload
def transpose(self, axes: None | Sequence[int], /) -> Self: ...
@overload
def transpose(self, *axes: int) -> Self: ...
```

Reply:
oh right, because `axes` cannot be passed as keyword.